### ISSUE TITLE
Body scroll is not blocked

### ISSUE THREAD

**Issue by alexanderKhvostov** [willmcpo/body-scroll-lock#206]
Hi, sorry but it doesn't work :( 
I still can sometime scroll body. Maybe Apple again has changed something?

Iphone 11pro IOs 13.6.
https://www.loom.com/share/9fe89fe098e64e64a26d5533f9921f29

**Comment by jvitela** [comment]
Hi, @alexanderKhvostov you have a look at this pull requests and see if it works for you:
https://github.com/willmcpo/body-scroll-lock/pull/207

**Comment by horaklukas** [comment]
I experience the same issue at least on iOS 12 😕 

The PR #207 is not helpful, settings `overflow: hidden` is not a solution.

**Comment by jvitela** [comment]
Hi @horaklukas, 

Can you please elaborate how did you test PR #207 ? Did you inspect the dom to make sure `overflow: hidden` was actually applied to the body element?

**Comment by yoonjaepark** [comment]
Hi @jvitela 

In some cases, the scroll has arrived at the bottom of the scroll element area. If there is a scroll in the content area, the scroll of the content moves.

<body>
  // scroll content
  <div>
  </div>
  // scroll popup element 
  <div>
  </div>
</body>

**Comment by JonesTwink** [comment]
@jvitela unfortunately this doesn't help. Since OP haven't provided much details I'd like to elaborate on when this bug might be seen. 
There're two conditions required for this bug to happen:
1. Safari's toolbar (which is located on the bottom of the screen) must be hidden
2.  User's finger must be placed on the bottom of the screen 
If this two conditions are met and user starts scrolling, it will happen.
If finger is being put on any other place of the screen or when the toolbar is shown, everything works as expected. 
I believe this bug is caused by some Safari internals and nothing can be done for now.

![ok](https://user-images.githubusercontent.com/15804937/110923266-c237b900-8331-11eb-90fa-6f9906bb7fca.png)
![ok](https://user-images.githubusercontent.com/15804937/110923281-c663d680-8331-11eb-9260-498f0cc89ddf.png)

**Comment by chendong-xie** [comment]
> @jvitela unfortunately this doesn't help. Since OP haven't provided much details I'd like to elaborate on when this bug might be seen.
> There're two conditions required for this bug to happen:
> 
> 1. Safari's toolbar (which is located on the bottom of the screen) must be hidden
> 2. User's finger must be placed on the bottom of the screen
> 
> If these two conditions are met and user starts scrolling, it will happen.
> 
> If finger is being put on any other place of the screen or when the toolbar is shown, everything works as expected.
> I believe this bug is caused by some Safari internals and nothing can be done for now.
> 
> ![](https://user-images.githubusercontent.com/15804937/110924453-1a22ef80-8333-11eb-9ee4-f736aacdc4ee.png)

hi，did you solve it?

**Comment by jvitela** [comment]
Hi @chendong-xie and @JonesTwink,

Yes, my [last change](https://github.com/willmcpo/body-scroll-lock/pull/207#issuecomment-815937919) fixes this issue. Setting `position:fixed` forces the browser to display the toolbar and therefore there is no scrollable area.

Please do test the latest changes of the [feature branch](https://github.com/jvitela/body-scroll-lock/tree/patch-1) and confirm if it also works for you. The examples are updated so you don't need to rebuild it yourself.

**Comment by chendong-xie** [comment]
hi @jvitela this way looks ok, thank you for your reply.
when you plan to release this feature?

**Comment by jvitela** [comment]
hi @chendong-xie, 

> when you plan to release this feature?

That question should be answered by @willmcpo

**Comment by n19htz** [comment]
> Hi @chendong-xie and @JonesTwink,
> 
> Yes, my [last change](https://github.com/willmcpo/body-scroll-lock/pull/207#issuecomment-815937919) fixes this issue. Setting `position:fixed` forces the browser to display the toolbar and therefore there is no scrollable area.
> 
> Please do test the latest changes of the [feature branch](https://github.com/jvitela/body-scroll-lock/tree/patch-1) and confirm if it also works for you. The examples are updated so you don't need to rebuild it yourself.

if you solving it with ```position: fixed``` you gotta be aware of scroll reset, i e for example you clicking a button that opens modal and that button is at middle of document (i e scrollTop > 0). Than when you lock the scroll with forcing safari to show bottom bar (by setting ```position:fixed```) your scroll is gonna reset (scrollTop === 0)

**Comment by jvitela** [comment]
@n19htz  Do you mean this? https://github.com/jvitela/body-scroll-lock/blob/3b7877b77c997751ff484a5f89ca2eeb5fda1bed/src/bodyScrollLock.js#L120

**Comment by n19htz** [comment]
> @n19htz Do you mean this? https://github.com/jvitela/body-scroll-lock/blob/3b7877b77c997751ff484a5f89ca2eeb5fda1bed/src/bodyScrollLock.js#L120

yes, i've tried 4.0.0-beta version and scroll restoration does not work there

**Comment by jvitela** [comment]
@n19htz Could you please provide a minimal reproducible example? or in case it happens with the included examples a video showing the issue would be great, and please indicate the browser and OS versions.

**Comment by n19htz** [comment]
https://user-images.githubusercontent.com/39006390/127184638-75b0282e-e99b-4f1a-9b32-56af224ec8a4.mp4

**Comment by pawelbujna** [comment]
I have a quetion, will this fix also fix the problem with jumping layout when clicking on inputs inside modals? I know that position: fixed casues the bug  it happens only on scrollable elements. To fix my case i had to completely hide the page contents and display only a modal. Feels like it could be part of this bug.

Example video: 

https://user-images.githubusercontent.com/7709957/127474305-2afbe204-73c1-4589-b370-6b4c0516c3a3.mov

**Comment by Trendymen** [comment]
``` javascript
const { scrollY, scrollX, innerHeight } = window;
    document.body.style.position = 'fixed'; 
    document.body.style.top =  -scrollY;   //Mabye there and next line should be add + "px" in the end.
    document.body.style.left = -scrollX; 
```

```javascript
  // Restore styles
    document.body.style.position = previousBodyPosition.position;
    document.body.style.top = previousBodyPosition.top;   //As the same above.
    document.body.style.left = previousBodyPosition.left;
```

**Comment by jvitela** [comment]
Hi @Trendymen  and @pawelbujna,

Please check https://github.com/willmcpo/body-scroll-lock/pull/229

**Comment by rick-liruixin** [comment]
They stopped the repairs. I had to do it myself, in the same way, with a new version of typeScript. And fix these problems for everyone to use.
```
npm i body-scroll-lock-upgrade
```
repair Changelog
Refer to the [releases](https://github.com/rick-liruixin/body-scroll-lock-upgrade/releases) page.
